### PR TITLE
[TGUI] Solar Control

### PR DESCRIPTION
**Pull request by RedWolfie**
## What Does This PR Do
This PR replaces Nano with TGUI for the Solar Control. It also refactors solar control a bit. 

## Why It's Good For The Game
Gets rid of nano

## Changelog
:cl:
add: Added TGUI for Solar Control
/:cl:
solar-demo1
solar-demo2
solar-demo3
solar-demo4

### CODE REVIEW

**Comment by SteelSlayer** [comment]
Judging from the screenshots, the power progress bar gets completely full at just 15,000 W. But the station's starting solar arrays have the capacity to go up to 90,000, and I think the bar actually has the potential to go up to 200,000 currently. Why does your UI not allow for numbers higher than 15,000?

**Comment by RedWolfie** [comment]
It depends on the amount of panels. Each panel has a maximum of 1500W. As I connected only 10 panels for testing, it maxes out at 15000

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Comment by SteelSlayer** [comment]
@RedWolfie Ah gotcha. That's cool that the max size scales with the amount of power you can have :+1:

File: code/modules/power/solar.dm [hunk 1]
```diff
@@ -334,15 +338,12 @@
...
-	switch(track)
-		if(1)
-			if(trackrate) //we're manual tracking. If we set a rotation speed...
-				cdir = targetdir //...the current direction is the targetted one (and rotates panels to it)
-		if(2) // auto-tracking
-			if(connected_tracker)
-				connected_tracker.set_angle(SSsun.angle)
-
-	set_panels(cdir)
+	if(TRACKER_TIMED && trackrate) //we're manual tracking. If we set a rotation speed...
```
[the remark this answers is not in the document]
Comment (RedWolfie) [reply code/modules/power/solar.dm]: fixed

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**